### PR TITLE
fix: Improve gateway app based on Demo app experience

### PR DESCRIPTION
**Pull request by sdsantos**

### CODE REVIEW

**Review by gnarea — COMMENTED** [review]
LGTM! It's great to see things generally worked, except for some minor tweaks here and there

File: app/build.gradle [hunk 1]
```diff
@@ -1,6 +1,7 @@
 buildscript {
     repositories {
         maven { url 'https://plugins.gradle.org/m2/' }
+        maven { url 'https://jitpack.io' }
```
Comment (gnarea) [inline app/build.gradle:4]: I think this was needed at one point but it doesn't seem to be needed anymore.

File: app/src/main/java/tech/relaycorp/gateway/data/preference/PublicGatewayPreferences.kt [hunk 2]
```diff
@@ -30,8 +34,11 @@ class PublicGatewayPreferences
 
     fun getCertificate() = { certificate }.toFlow()
         .map {
-            if (it.isEmpty()) throw RuntimeException("Public Gateway Certificate not set")
-            val certificateBytes = it.toByteArray(Charset.defaultCharset())
+            val certificateBytes = if (it.isEmpty()) {
+                resources.openRawResource(R.raw.public_gateway_cert).readBytesAndClose()
+            } else {
```
Comment (gnarea) [inline app/src/main/java/tech/relaycorp/gateway/data/preference/PublicGatewayPreferences.kt:39]: FYI: This is OK for now, but we'll probably have to set all the public gateway data upfront, when the app is started for the first time.

**Review by gnarea — CHANGES_REQUESTED** [review]

File: app/build.gradle (hunk 1 above)
Comment (sdsantos) [reply app/build.gradle]: Done

**Review by gnarea — APPROVED** [review]

**Pull request merged**